### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by simplegoose**
In this PR, the following was done:

- A new view file under the posts directory was created for handling post creation on the client side.
- A new controller method was created for creating the post on the database.
- A new controller likes_controller was created for handling like actions.
- A new form was created on the show.html.rb file for handling like actions on a post.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -4,6 +4,28 @@ def index
...
+  def new
+    post = Post.create
+    respond_to do |format|
+      format.html { render :new, locals: { post: } }
+    end
+  end
+
+  def create
+    user = current_user
+    post = Post.new(params.require(:post).permit(:title, :text))
```
Comment (franclobo) [inline app/controllers/posts_controller.rb]: - **[OPTIONAL]** Please add the params method as private for a clean code.  [Lesson link about the strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

File: app/views/posts/new.html.erb
Comment (franclobo) [inline app/views/posts/new.html.erb:1]: - [x] Well done creating the form for a new post, but how can I access to it? Please add a link in the name of the user to open the user#show page after that when the user clicks on the See all posts button it shows the complete list of the posts from the current user, kindly add a button to create a new post at the end of this page (post#index). Finally, when the user opens the post#show page it is possible to add likes, please 🙏🏽 add a button to create new comments to be able to add new comments to the post.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @simplegoose, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Use preprocessed HTML file with embedded Ruby code. ✔️
- Use layouts and templates for shared content. ✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly add the comments controller file. It should have new and create actions to be able to create new comments.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+        <div class="alert_inner alert-success">
+          <%= msg %>
+        </div>
+      <% end %>
+    </div>
+  <% end %>
   <h1>Users</h1>
+  <div class="create_post">
+    <%= link_to "Create Post", "/post/new" %>
+  </div>
```
Comment (Qoosim) [inline app/views/users/index.html.erb]: - [ ] This is not supposed to be here. Kindly remove it from here.

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
   <% @users.each do |user| %>
     <div class="user_info_wrapper">
       <%= image_tag user.photo, :class => 'user_avatar' %>
```
Comment (Qoosim) [inline app/views/users/index.html.erb]: - [x] Great job implementing the required changes from the previous reviewer. However, there're some misconceptions on the recommendations from the previous reviewer. This is how it should work.
- [x] 1.  All the users added to the db with `rails console` should get displayed when the app starts. Each user name should be a link to the list of user's posts and a form to create new posts.
- [x] Each post should be a link to the post show page where there would be a form to add comments to the posts and the posts could be liked.
- [x] Kindly follow the wirefram from the project requirements to implement all this. You would not have any issues if you follow the wireframe properly.

File: config/routes.rb [hunk 4]
```diff
@@ -3,8 +3,17 @@
...
+  get "users/:user_id/posts/:post_id", to: "posts#show", as: "posts_show"
+
+  get "post/new", to: "posts#new", as: "new_post" # new
+  post "post/create", to: "posts#create", as: "post_create" #create
+
+  post "like/create", to: "likes#create", as: "like_create" #create
+
+  get "users/:user_id/post/:post_id/comment/new", to: "comments#new" #new
+  post "comment/create", to: "comments#create", as: "comment_create" #create
+
```
Comment (Qoosim) [inline config/routes.rb]: - [OPTIONAL] Configure your routes this way would make your routes clean and easier to read.

```
   resources :users, only: [:index, :show] do
       resources :posts, only: [:index, :show]
   end
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @simplegoose, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  Implmented some required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 2 above)
Comment (simplegoose) [reply app/views/users/index.html.erb]: Kindly, where should this button be @Qoosim. I am confused at this point.

File: config/routes.rb [hunk 5]
```diff
@@ -3,8 +3,17 @@
...
+    resources :likes, only: [:create]
+    resources :comments, only: [:create]
+  end
+
   get "users/:user_id/posts", to: "posts#index"
-  get "users/:user_id/posts/:post_id", to: "posts#show"
+  get "users/:user_id/posts/:post_id", to: "posts#show", as: "posts_show"
+
+  get "users/:user_id/posts/:post_id/comment/new", to: "comments#new" #new
+
```
Comment (christianbiring1) [inline config/routes.rb:18]: [OPTIONAL]
- [ ] As suggested by the previous reviewer, This `route.rb` can be refactored further like so:
```rb
resources :users, only: [:index, :show] do
  resources :posts, only: [:new, :create, :show]
 # ...
end
```

**Review by christianbiring1 — APPROVED** [review]
Hi @simplegoose , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- [ ] Great work on implementing the changes as suggested by the previous reviewer :100: 
- [ ] The app is running correctly and all the URL are well set :+1: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**